crates/re_types_builder/src/codegen/rust/api.rs

```diff
@@ -768,16 +768,20 @@ fn quote_trait_impls_from_obj(
                 objects: &Objects,
                 extras: impl IntoIterator<Item = String>,
             ) -> (usize, TokenStream) {
-                let components = iter_archetype_components(obj, attr)
+                let mut components = iter_archetype_components(obj, attr)
                     .map(|fqname| {
                         objects[fqname.as_str()]
                             .try_get_attr::<String>(crate::ATTR_RERUN_LEGACY_FQNAME)
                             .unwrap_or(fqname)
                     })
                     .chain(extras)
                     .collect::<Vec<_>>();
+                components.sort();
+                components.dedup();
```

Comment:
nit: could be shorter and clearer with

```suggestion
                    .collect::<BTreeSet<_>>();
```